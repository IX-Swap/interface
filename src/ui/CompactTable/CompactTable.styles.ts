import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  cell: {
    paddingBottom: theme.spacing(1.5),
    paddingRight: 0,
    paddingTop: 8
  },
  card: {
    backgroundColor: theme.palette.background.paper,
    padding: theme.spacing(3),
    borderRadius: 8
  },
  columns: {
    display: 'grid',
    gridTemplateColumns: '1fr 1fr',
    rowGap: theme.spacing(3)
  },
  label: {
    color: theme.palette.text.secondary
  },
  value: {
    fontSize: '14px'
  },
  iconButton: {
    background: theme.palette.button.bgLight
  }
}))
