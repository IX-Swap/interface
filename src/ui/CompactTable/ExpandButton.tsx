import { Button, Grid } from '@mui/material'
import React, { useContext } from 'react'
import { Icon } from 'ui/Icons/Icon'
import { useStyles } from 'ui/CompactTable/ExpandButton.styles'
import { ActiveElementContext } from 'app/context/ActiveElementContextWrapper'
import { Serialized } from 'types/base'

export interface ExpandButtonProps<T extends Serialized> {
  item: T
}

export const ExpandButton = <T extends Serialized>({
  item
}: ExpandButtonProps<T>) => {
  const context = useContext(ActiveElementContext)
  const handleClick = () => {
    context?.toggleRow(item._id)
  }
  const classes = useStyles()
  return (
    <Grid item xs={12}>
      <Button
        onClick={() => handleClick()}
        fullWidth
        className={classes.iconButton}
      >
        <Icon name='more-horizontal' />
      </Button>
    </Grid>
  )
}
