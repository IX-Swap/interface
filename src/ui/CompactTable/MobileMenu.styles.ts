import makeStyles from '@mui/styles/makeStyles'

export const useStyles = makeStyles(theme => ({
  drawer: {
    padding: theme.spacing(2.5),
    borderRadius: '8px 8px 0 0',
    backgroundColor: theme.palette.background.paper
  },
  titleRow: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: theme.spacing(3)
  },
  header: {
    color: theme.palette.mode === 'light' ? '#666666' : '#ffffff',
    lineHeight: 1.5
  },
  tableRow: {
    borderBottom: 'none'
  },
  rowBox: {
    backgroundColor: theme.palette.mode === 'light' ? '#F6F4FD' : '#494166',
    borderRadius: theme.spacing(0.5),
    padding: theme.spacing(1.25)
  }
}))
