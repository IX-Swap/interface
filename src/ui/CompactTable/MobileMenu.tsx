import { Box, Drawer, Grid, IconButton, Typography } from '@mui/material'
import { ActiveElementContext } from 'app/context/ActiveElementContextWrapper'
import { useStyles } from 'ui/CompactTable/MobileMenu.styles'
import { useAppBreakpoints } from 'hooks/useAppBreakpoints'
import React, { useContext, useMemo } from 'react'
import { Serialized } from 'types/base'
import { Icon } from 'ui/Icons/Icon'

export const MobileMenu = <T extends Serialized>({
  items,
  titleExtractor,
  actions
}: {
  items: T[]
  titleExtractor: (item: T) => string
  actions: (item: T) => React.ReactElement
}) => {
  const context = useContext(ActiveElementContext)
  const open = context?.hasOpenIndices
  const classes = useStyles()
  const openIndex = context?.openIndex
  const { isTablet } = useAppBreakpoints()

  const selectedItem = useMemo(
    () => items.filter(item => item._id === openIndex)?.[0],
    [openIndex, items]
  )
  console.log({ items, openIndex })
  if (selectedItem === undefined) {
    return null
  }
  const onClose = () => {
    if (openIndex !== undefined) {
      context?.toggleRow(openIndex)
    }
  }

  return (
    <>
      {isTablet && (
        <Drawer
          anchor='bottom'
          open={open}
          onClose={onClose}
          PaperProps={{ sx: { backgroundColor: 'transparent' } }}
        >
          <Box className={classes.drawer}>
            <Box className={classes.titleRow}>
              <Typography
                variant='subtitle2'
                fontWeight={600}
                className={classes.header}
              >
                {titleExtractor(selectedItem)}
              </Typography>
              <IconButton onClick={onClose} size='medium'>
                <Icon name='close' />
              </IconButton>
            </Box>
            <Grid flexDirection='column'>
              <Grid display='flex' flexDirection={'column'} gap={2}>
                {actions(selectedItem)}
              </Grid>
              <Box mb={2} />
            </Grid>
          </Box>
        </Drawer>
      )}
    </>
  )
}
