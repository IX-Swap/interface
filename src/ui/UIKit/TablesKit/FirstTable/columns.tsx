// import {
//   renderBalance,
//   renderUSDPrice,
//   renderCurrencyLabel
// } from 'ui/UIKit/TablesKit/utils/utils'
import { TableColumn } from 'types/util'
import { FirstTableItem } from 'ui/UIKit/TablesKit/FirstTable/FirstTable'
import {
  renderBalance,
  renderUSDPrice,
  renderCurrencyLabel
} from 'helpers/tables'

export const columns: Array<TableColumn<FirstTableItem>> = [
  {
    key: 'currency',
    label: '',
    render: renderCurrencyLabel
  },
  {
    label: 'Balance',
    key: 'balance',
    render: renderBalance
  },
  {
    label: 'USD Value',
    key: 'usdValue',
    render: renderUSDPrice
  },
  {
    label: 'AvailableBalance',
    key: 'availableBalance',
    render: renderUSDPrice
  }
]
