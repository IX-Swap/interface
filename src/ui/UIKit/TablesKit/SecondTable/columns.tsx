import { TableColumn } from 'types/util'
import { RenderBolderText, renderSGDPrice } from 'helpers/tables'
import { SecondTableItem } from 'ui/UIKit/TablesKit/SecondTable/SecondTable'

export const columns: Array<TableColumn<SecondTableItem>> = [
  {
    key: 'pair',
    label: 'Pair',
    render: RenderBolderText
  },
  {
    label: 'Symbol',
    key: 'symbol',
    render: RenderBolderText
  },
  {
    label: 'Issued By',
    key: 'issuedBy',
    render: RenderBolderText
  },
  {
    label: 'Type',
    key: 'type'
  },
  {
    label: 'Price',
    key: 'price',
    render: renderSGDPrice
  }
]
