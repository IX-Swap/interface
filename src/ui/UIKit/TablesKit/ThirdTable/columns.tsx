import { TableColumn } from 'types/util'
import { ThirdTableItem } from 'ui/UIKit/TablesKit/ThirdTable/ThirdTable'
import { RenderBolderText, renderStatus } from 'helpers/tables'

export const columns: Array<TableColumn<ThirdTableItem>> = [
  {
    key: 'currency',
    label: '',
    render: RenderBolderText
  },
  {
    label: 'Bank Name',
    key: 'bankName',
    render: RenderBolderText
  },
  {
    label: 'Account Number',
    key: 'accountNumber'
  },
  {
    label: 'SwiftCode',
    key: 'swiftCode'
  },
  {
    label: 'Status',
    key: 'status',
    headAlign: 'right',
    align: 'right',
    render: renderStatus
  }
]
