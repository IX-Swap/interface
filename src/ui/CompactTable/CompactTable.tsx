import { Button, Grid, TableBody, TableCell, TableRow } from '@mui/material'
import { ActiveElementContext } from 'app/context/ActiveElementContextWrapper'
import { useStyles } from 'ui/CompactTable/CompactTable.styles'
import { CompactRowProps } from 'components/TableWithPagination/CompactRow'
import get from 'lodash/get'
import React, { useContext } from 'react'
import { Serialized } from 'types/base'
import { TableColumn } from 'types/util'
import { Icon } from 'ui/Icons/Icon'
import { TableViewRendererProps } from 'ui/UIKit/TablesKit/components/TableView/TableView'

export interface CompactBodyProps<T extends Serialized>
  extends TableViewRendererProps<T> {
  renderRow?: (props: CompactRowProps<T>) => JSX.Element
  menu: React.ReactElement
}

export interface RenderCellProps<T extends Serialized>
  extends TableColumn<T, string> {
  item: T
}
export const renderCell = <T extends Serialized>({
  render,
  key,
  item
}: RenderCellProps<T>) => {
  return (
    key.length > 0 &&
    (typeof render === 'function'
      ? render(get(item, key), item)
      : get(item, key))
  )
}

export const CompactTable = <T extends Serialized>(
  props: CompactBodyProps<T>
) => {
  const { columns, items, menu } = props
  const classes = useStyles()
  const context = useContext(ActiveElementContext)
  const handleClick = (item: T) => {
    context?.toggleRow(item._id)
  }

  return (
    <>
      <TableBody>
        {items.map((item, i) => (
          <TableRow key={i}>
            <TableCell className={classes.cell}>
              <Grid
                container
                flexDirection='row'
                rowGap={3}
                className={classes.card}
              >
                <Grid
                  item
                  container
                  rowGap={1}
                  className={classes.columns}
                  xs={12}
                >
                  {columns.map(({ label, key, render }, index) => (
                    <Grid
                      item
                      container
                      key={key}
                      flexDirection='row'
                      rowGap={1}
                    >
                      <Grid item xs={12} className={classes.label}>
                        {label}
                      </Grid>
                      <Grid item xs={12} className={classes.value}>
                        {renderCell({ render, key, item, label })}
                      </Grid>
                    </Grid>
                  ))}
                </Grid>
                <Grid item xs={12}>
                  <Button
                    onClick={() => handleClick(item)}
                    fullWidth
                    className={classes.iconButton}
                  >
                    <Icon name='more-horizontal' />
                  </Button>
                </Grid>
              </Grid>
            </TableCell>
          </TableRow>
        ))}
      </TableBody>
      {menu}
    </>
  )
}
