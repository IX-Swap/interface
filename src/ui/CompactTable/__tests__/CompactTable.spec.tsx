import { ActiveElementContextWrapper } from 'app/context/ActiveElementContextWrapper'
import { compactColumns } from 'app/pages/accounts/pages/banks/pages/BanksList/columns'
import { banksQueryKeys } from 'config/queryKeys'
import React from 'react'
import { render } from 'test-utils'
import { bank } from '__fixtures__/authorizer'
import { user } from '__fixtures__/user'
import { CompactTable } from '../CompactTable'

describe('CompactTable', () => {
  const props = {
    columns: compactColumns,
    items: [bank],
    loading: false,
    hasActions: false,
    cacheQueryKey: banksQueryKeys.getListByUserId(user._id)
  }

  it('compact table matches snapshot', () => {
    const { container } = render(
      <ActiveElementContextWrapper>
        <CompactTable {...props} menu={<></>} />
      </ActiveElementContextWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
