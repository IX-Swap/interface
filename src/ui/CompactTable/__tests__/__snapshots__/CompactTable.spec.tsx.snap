// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactTable compact table matches snapshot 1`] = `
<div>
  <tbody
    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
  >
    <tr
      class="MuiTableRow-root css-8dfetm-MuiTableRow-root"
    >
      <td
        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium makeStyles-cell-1 css-17e0zod-MuiTableCell-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container makeStyles-card-2 css-1mjjv40-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 makeStyles-columns-3 css-1nk7a39-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-az2snp-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-label-4 css-ewf807-MuiGrid-root"
              >
                Currency
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-value-5 css-ewf807-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-70qvj9"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1zdpvo-MuiTypography-root"
                  >
                    SGD
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-az2snp-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-label-4 css-ewf807-MuiGrid-root"
              >
                Bank Name
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-value-5 css-ewf807-MuiGrid-root"
              >
                Rocketbank
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-az2snp-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-label-4 css-ewf807-MuiGrid-root"
              >
                Account Number
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-value-5 css-ewf807-MuiGrid-root"
              >
                1234567890
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-az2snp-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-label-4 css-ewf807-MuiGrid-root"
              >
                Account Name
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-value-5 css-ewf807-MuiGrid-root"
              >
                Oleg Tinkoff
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item css-az2snp-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-label-4 css-ewf807-MuiGrid-root"
              >
                Swift Code
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 makeStyles-value-5 css-ewf807-MuiGrid-root"
              >
                SWIFTCODE
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth MuiButtonBase-root makeStyles-iconButton-7 css-ns7bdg-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                class="css-1s8o7ec"
                fill="#778194"
                height="24px"
                width="24px"
              >
                <use
                  href="image-mock#more-horizontal"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </td>
    </tr>
  </tbody>
</div>
`;
